app/src/main/java/explorer/city/com/cityexplorer/viewModel/ViewCityViewModel.kt

```diff
@@ -0,0 +1,94 @@
+package explorer.city.com.cityexplorer.viewModel
+
+import android.arch.lifecycle.MutableLiveData
+import android.arch.lifecycle.ViewModel
+import android.util.Log
+import explorer.city.com.cityexplorer.model.CityInfo
+import explorer.city.com.cityexplorer.model.CityScore
+import explorer.city.com.cityexplorer.model.Photos
+import explorer.city.com.cityexplorer.network.TeleportService
+import io.reactivex.Single
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+class ViewCityViewModel: ViewModel() {
+
+    private val TAG = "ViewCityViewModel"
+
+    private val subscriptions by lazy { CompositeDisposable() }
+    val liveCityInfo: MutableLiveData<CityInfo> by lazy { MutableLiveData<CityInfo>() }
+    val liveImageInfo: MutableLiveData<Photos> by lazy { MutableLiveData<Photos>() }
+    val liveCityScore: MutableLiveData<CityScore> by lazy { MutableLiveData<CityScore>() }
+
+    fun getCityInfo(cityLink: String ): MutableLiveData<CityInfo> {
+        if (shouldFetchData()) {
+            fetchCityInfo(cityLink)
+        }
+        return liveCityInfo
+    }
+
+    private fun shouldFetchData(): Boolean {
+        return liveCityInfo.value == null
+                && liveImageInfo.value == null
+                && liveCityScore.value == null
+    }
+
+    private fun fetchCityInfo(cityLink: String) {
+        val sub = cityInfoSingle(cityLink)
+                .observeOn(AndroidSchedulers.mainThread())
+                .doOnSuccess { liveCityInfo.value = it }
+                .flatMap { cityInfoSingle(it.links.urbanArea.href) }
+                .subscribe(
+                        {
+                            fetchCityScore(it.links.scores.href)
+                            fetchImageInfo(it.links.image.href)
+                        },
+                        { Log.e(TAG, "Error -> $it") }
+                )
+
+        subscriptions.add(sub)
+    }
+
+    private fun fetchCityScore(cityInfoUrl: String) {
+        val sub = cityScoreInfoSingle(cityInfoUrl)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        { liveCityScore.value = it },
+                        { Log.e(TAG, "Error -> $it") }
+                )
+
+        subscriptions.add(sub)
+    }
+
+    private fun fetchImageInfo(cityInfoUrl: String) {
+        val sub = cityImageSingle(cityInfoUrl)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                        { liveImageInfo.value = it },
+                        { Log.e(TAG, "Error -> $it") }
+                )
+
+        subscriptions.add(sub)
+    }
+
+    private fun cityInfoSingle(url: String): Single<CityInfo> {
```

Comment:
This is a poor naming scheme. Why not `getCityInfo`?

What if you change this from returning a `Single` to something else? At the point you'll have to change the method name. There are plenty of other functions here that are poorly named.